src/Extension/Traits/FluentAdminTrait.php

```diff
@@ -566,4 +577,80 @@ protected function inEveryStage($doSomething)
             });
         }
     }
+
+    /**
+     * Update save action based on localisation state
+     * valid states:
+     * - localised (localise label)
+     * - not localised (save label)
+     *
+     * @param FieldList $actions
+     * @param DataObject $record
+     */
+    protected function updateSaveAction(FieldList $actions, DataObject $record)
+    {
+        if ($record->hasExtension(Versioned::class)) {
+            return;
+        }
+
+        if ($record->existsInLocale()) {
+            // keep the action unchanged as the record is localised
+            return;
+        }
+
+        $saveAction = $actions->fieldByName('MajorActions.action_doSave');
```

Comment:
This won't work on gridfield save action though will it? It'll only for for sitetree / cmsmain actions. Is it possible to change this search to match doSave action at any level?